packages/components/src/utils/useAnimationState.ts

```diff
@@ -105,6 +110,21 @@ export const useAnimationState = (
     }
   }, [isOpen, timingEnter, timingExit, state])
 
+  const hasBeenOpenRef = useRef(false)
+
+  useEffect(() => {
+    if (state === 'entered' && onAfterEntered) onAfterEntered()
+
+    if (state === 'exited' && onAfterExited) {
+      if (hasBeenOpenRef.current) {
+        onAfterExited()
+      }
+      hasBeenOpenRef.current = false
+    } else {
+      hasBeenOpenRef.current = true
+    }
+  }, [state, onAfterExited, onAfterEntered])
+
```

Comment:
Looks like the "changing callbacks" issue demonstrated [here](https://github.com/looker-open-source/components/pull/2842#discussion_r715142617) is still an issue for `onAfterOpen`. Just need a slight tweak to fix it:
```suggestion
  const previousStateRef = useRef(state)

  useEffect(() => {
    if (state === 'entered' && previousStateRef.current !== 'entered') {
      onAfterEntered?.()
    }

    if (state === 'exited' && previousStateRef.current !== 'exited') {
      onAfterExited?.()
    }
    previousStateRef.current = state
  }, [state, onAfterExited, onAfterEntered])

```